src/values.rst

```diff
@@ -95,6 +95,9 @@ The :t:`type specification` of a :t:`constant` shall have ``'static``
 The :t:`type` of a :t:`constant` shall implement the :std:`core::marker::Sized`
 :t:`trait`.
 
+:dp:`fls_ox6sgl9n43g2`
+The type of a :t:`constant` cannot be a :t:`mutable reference type`.
```

Comment:
This doesn't cover it.

```rust
struct W<T>(T);
const _: W<&mut ()> = W(&mut ()); //~ ERROR
const _: W<&AtomicU8> = W(&AtomicU8::new(0)); //~ ERROR
```

Reply:
to be clear, are those additional rules

Reply:
What I'm getting at here specifically is that the final value can't contain any mutable references, not just that it can't be a mutable reference type.

```rust
struct W<T>(T);
static mut S: u8 = 0;
const _: W<&mut u8> = W(unsafe { &mut S }); //~ ERROR
//       ^^^^^^^^^^ Not a mutable reference type.
```

And then, yes, there are additional rules that prohibit mutable borrows of lifetime-extended temporaries even when that doesn't result in a mutable reference being contained in the final value.

```rust
struct W<T>(T);
const _: W<&u8> = W(&mut 0); //~ ERROR
```